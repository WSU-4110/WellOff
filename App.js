/* eslint-disable prettier/prettier */
import React from 'react';
import {Text, View} from 'react-native';
import Ionic from 'react-native-vector-icons/FontAwesome';
import {NavigationContainer} from '@react-navigation/native';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import HomeScreen from './screens/HomeScreen';
import MyPlanScreen from './screens/MyPlanScreen';
import SettingsScreen from './screens/SettingsScreen';
import HealthScreen from './screens/MyPlanScreens/HealthScreen';
import RemindersScreen from './screens/MyPlanScreens/RemindersScreen'
import ProfileScreen from './screens/MyPlanScreens/EditProfile'
import PreferencesScreen from './screens/MyPlanScreens/EditPreferences'

const Tab = createBottomTabNavigator();
const Stack = createNativeStackNavigator();

function HomeTabs() {
  return (
    <Tab.Navigator screenOptions={({route}) => ({
      tabBarShowLabel: false,
      tabBarStyle: {
        backgroundColor: '#000000',
        height: 80,
      },
      headerShown: false,
      tabBarIcon: ({focused, color, size}) => {
        let iconName;
        if (route.name === 'Home') {
          iconName = focused ? 'ios-home' : 'ios-home-outline';
        } else if (route.name === 'My Plan') {
          iconName = focused ? 'ios-home' : 'ios-home-outline';
        } else if (route.name === 'Settings') {
          iconName = focused ? 'ios-home' : 'ios-home-outline';
        } else if (route.name === 'Reminders') {
          iconName = focused ? 'ios-home' : 'ios-home-outline';
        } else if (route.name === 'Health') {
          iconName = focused ? 'ios-home' : 'ios-home-outline';
        }
        return <Ionic name={iconName} size={size} color={color} />;
      },
    })}>
      <Tab.Screen name="Home" component={HomeScreen} />
      <Tab.Screen name="My Plan" component={MyPlanScreen} />
      <Tab.Screen name="Settings" component={SettingsScreen} />
    </Tab.Navigator>
  );
}

const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={({route}) => ({      
        headerShown: true,
        })}>
        <Stack.Screen name="Home" component={HomeTabs} />
        <Stack.Screen name="Profile" component={ProfileScreen} />
        <Stack.Screen name="Preferences" component={PreferencesScreen} />
        <Stack.Screen name="Health" component={HealthScreen} />
        <Stack.Screen name="Notification" component={RemindersScreen} />

      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
