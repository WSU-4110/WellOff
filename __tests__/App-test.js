/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import {
  __getCompletedIcon,
  __getDescriptionStyle,
  clickEventListener,
  onClickListener,
  __renderColors,
  __renderImages,
} from '../TestCasses';

test('get icons', () => {
  const icon = 'https://img.icons8.com/flat_round/64/000000/delete-sign.png';
  expect(__getCompletedIcon(2)).toEqual(icon);
});

test('geting styles', () => {
  const style = {
    textDecorationLine: 'line-through',
    fontStyle: 'italic',
    color: '#808080',
  };
  expect(__getDescriptionStyle(1)).toEqual(style);
});

test('click EventListner', () => {
  expect(clickEventListener()).toBeUndefined();
});

test('render colors', () => {
  const tree = __renderColors();
  expect(tree).toMatchSnapshot();
});

test('render Images', () => {
  const tree = __renderColors();
  expect(tree).toMatchSnapshot();
});

test('Button pressed', () => {
  expect(onClickListener()).toBeUndefined();
});
