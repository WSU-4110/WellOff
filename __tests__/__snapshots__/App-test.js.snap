// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Images 1`] = `
<View
  style={
    Object {
      "color": "white",
      "flexDirection": "row",
    }
  }
>
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#00BFFF",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#FF1493",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#00CED1",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#228B22",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#20B2AA",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#FF4500",
        },
      ]
    }
  />
</View>
`;

exports[`render colors 1`] = `
<View
  style={
    Object {
      "color": "white",
      "flexDirection": "row",
    }
  }
>
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#00BFFF",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#FF1493",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#00CED1",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#228B22",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#20B2AA",
        },
      ]
    }
  />
  <TouchableOpacity
    style={
      Array [
        Object {
          "borderRadius": 40,
          "height": 40,
          "marginHorizontal": 3,
          "width": 40,
        },
        Object {
          "backgroundColor": "#FF4500",
        },
      ]
    }
  />
</View>
`;
